src/main/java/com/google/cloud/discotoproto3converter/proto3/Field.java

```diff
@@ -19,31 +19,35 @@
 import java.util.List;
 import java.util.Objects;
 
-public class Field extends ProtoElement {
-  private final String name;
+// compareTo() == 0 and equals() are inconsistent for this implementation
+public class Field extends ProtoElement<Field> {
   private Message valueType;
   private final boolean repeated;
   private final boolean optional;
   private Message keyType;
   private final List<Option> options = new ArrayList<>();
+  private final boolean firstInOrder;
```

Comment:
It might be nice to explain why you need this here. At first glance, it seems to break an abstraction: why should a field know whether it's first in a collection? That seems to be a property of the collection...